tests/unit/core/api/test_index.js

```diff
@@ -469,7 +469,7 @@ describe(__filename, () => {
 
       sinon.assert.calledWith(
         _log.warn,
-        'Response from API was not JSON (was Content-Type: [unknown])',
+        'Response from API was not JSON (was Content-Type: null)',
```

Comment:
This looks like a problem because `[unknown]` comes from our code, not from the lib.

Reply:
It looks like the issue is that [`generateHeaders`](Content-Type) is now creating headers that convert `null` or `undefined` into strings. So if we call `generateHeaders({ 'Content-Type': null })`, what we actually get is a header where `Content-Type` is `'null'` as opposed to `null`. Prior to this patch, we would have gotten an actual `null` in the `Content-Type`.

The code for `generateHeaders` simply calls `return new Headers(headerData)`, so it's the `Headers` class that is doing this. I guess because of the `isomorphic-fetch` polyfill, the `Headers` class behaviour has changed? Or maybe I'm misinterpreting this whole thing?

The only difference between this version of `isomorphic-fetch` and the previous one is that it has updated its dependencies; `node-fetch` went from `^1.0.1` to `^2.6.1` and `whatwg-fetch` went from `>=0.10.0` to `^3.4.1`. So I guess it's one of those libraries that is responsible for this change.

If that's the case, then does that mean that the new behaviour is the expected behaviour? Also, are we concerned about this behaviour, and/or do we need to change any of our code because of it?

Reply:
This change is actually documented here: https://github.com/node-fetch/node-fetch/blob/master/docs/v2-UPGRADE-GUIDE.md#headers

So I guess the new behaviour is the correct behaviour and we should adjust our code to expect it.

Reply:
@willdurand I made a different change to the test which allows it to pass now, and I think this change makes more sense given the above.

Reply:
Great, thanks for the extra input on this. I guess we don't need to have `|| 'unknown'` anymore but maybe it's fine to keep them around. I am wondering if we shouldn't document why it's `null` and not `unknown` in the test case (which I think mentions "unknown" in the test desc)

Reply:
I figured because `Content-Type` can still be `''`, we should keep the logic for `unknown`, otherwise the message will look like "Response from API was not JSON (was Content-Type:)". Maybe we don't care about that?  It's really just about `Content-Type` not being `application/json`, right? I'll change the name of the test to reflect that.

Reply:
yeah. If we don't have a test case for empty `content-type`, I suggest adding one to cover the `|| 'unknown'` par.

Reply:
Nevermind. I don't see the point in keeping the `|| '[unknown]'` logic. I've removed it and changed the test case.